code/game/machinery/jukebox.dm

```diff
@@ -1,55 +1,143 @@
-//This file was auto-corrected by findeclaration.exe on 25.5.2012 20:42:32
+//
+// Media Player Jukebox
+// Rewritten by Leshana from existing Polaris code, merging in D2K5 and N3X15 work
+//
 
-/datum/track
-	var/title
-	var/track
-
-/datum/track/New(var/title, var/track)
-	src.title = title
-	src.track = track
-
-/datum/track/proc/GetTrack()
-	if(ispath(track, /music_track))
-		var/music_track/music_track = decls_repository.get_decl(track)
-		return music_track.song
-	return track
+#define JUKEMODE_NEXT        1 // Advance to next song in the track list
+#define JUKEMODE_RANDOM      2 // Not shuffle, randomly picks next each time.
+#define JUKEMODE_REPEAT_SONG 3 // Play the same song over and over
+#define JUKEMODE_PLAY_ONCE   4 // Play, then stop.
 
 /obj/machinery/media/jukebox/
 	name = "space jukebox"
 	icon = 'icons/obj/jukebox.dmi'
 	icon_state = "jukebox2-nopower"
 	var/state_base = "jukebox2"
-	anchored = TRUE
-	density = TRUE
```

Comment:
why switch off of these?

Reply:
Because he decided to put those on line 16 and 17 instead (Might be how VG does it?).
Makes no difference but still weird to use 1 and 0 over TRUE and FALSE in setting Booleans in my opinion.